soc/arm/silabs_exx32/efr32mg24/Kconfig.defconfig.series

```diff
@@ -18,6 +18,8 @@ config NUM_IRQS
 
 config PM
 	default n
+	select COUNTER
+	select UART_INTERRUPT_DRIVEN
```

Comment:
Hmm, we actually should be able to use PM without UART, so this needs to be reworked

Reply:
Yes, it would be nice to fix this for both platforms. We missed this in #55997 but the fix shouldn't be hard.

My understanding is that if you do use UART and PM at the same time, UART needs to be interrupt driven (otherwise it needs to keep polling so it cannot really go to sleep).

However in real life apps it's not uncommon to completely disable UART, that's why I think this needs to be reformulated - i.e. `UART_INTERRUPT_DRIVEN` should be required only if both UART and PM are enabled.
What was introduced in #55997 can be fixed in a separate PR I think.

Worst case we can completely remove this line and fix this separately or require users to know what they're doing if they use polling and PM at the same time

Reply:
You are right - thinking of custom boards, the UART may be disabled, which currently results in a Kconfig dependency error. Would you agree with this change that fixes the issue:

```diff
 config PM
 	default n
 	select COUNTER
-	select UART_INTERRUPT_DRIVEN
+	select UART_INTERRUPT_DRIVEN if SERIAL_SUPPORT_INTERRUPT
```